crates/pubsub/src/implementors/mosquitto.rs

```diff
@@ -0,0 +1,157 @@
+use std::sync::{Arc, Mutex};
+
+use anyhow::{Context, Result};
+use async_channel::Receiver;
+use futures::executor::block_on;
+use mosquitto_rs::{Client, Message, QoS};
+use slight_common::BasicState;
+
+#[derive(Clone)]
+pub struct MosquittoImplementor {
+    mqtt: Arc<Mutex<Client>>,
+    host: String,
+    port: i32,
+    subscriptions: Arc<Mutex<Vec<String>>>,
+    subscriber: Arc<Mutex<Option<Receiver<Message>>>>,
+}
+
+impl std::fmt::Debug for MosquittoImplementor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "MosquittoImplementor")
```

Comment:
Please add a TODO or FIXME here to remind us that we need to add a better debugging message to `fmt`. It is better to show the `mqtt` client struct.

Reply:
~will do this.

Reply:
done ✅